src/System.Security.Cryptography.Xml/tests/KeyInfoTest.cs

```diff
@@ -195,18 +196,139 @@ public void Complex()
+                    Assert.Equal(Convert.FromBase64String(dsaQ), dsaParams.Q);
+                    Assert.Equal(Convert.FromBase64String(dsaG), dsaParams.G);
+                    Assert.Equal(Convert.FromBase64String(dsaY), dsaParams.Y);
+
+                    // J is an optimization it should either be null or correct value
+                    if (dsaParams.J != null)
+                    {
+                        Assert.Equal(Convert.FromBase64String(dsaJ), dsaParams.J);
+                    }
+
+                    Assert.Equal(Convert.FromBase64String(dsaSeed), dsaParams.Seed);
+
+                    byte[] counter = Convert.FromBase64String(dsaPgenCounter);
+                    Assert.InRange(counter.Length, 1, 4);
+                    int counterVal = 0;
+                    for (int j = 0; j < counter.Length; j++)
+                    {
+                        counterVal <<= 8;
+                        counterVal |= counter[j];
+                    }
+
+                    Assert.Equal(counterVal, dsaParams.Counter);
+                }
+                else if (clause is RSAKeyValue)
+                {
+                    pathsCovered |= 1 << 2;
+
+                    var rsaKV = clause as RSAKeyValue;
+                    RSA rsaKey = rsaKV.Key;
+                    RSAParameters rsaParameters = rsaKey.ExportParameters(false);
+
+                    Assert.Equal(Convert.FromBase64String(rsaModulus), rsaParameters.Modulus);
+                    Assert.Equal(Convert.FromBase64String(rsaExponent), rsaParameters.Exponent);
+                }
+                else if (clause is KeyInfoNode)
+                {
+                    pathsCovered |= 1 << 3;
+
+                    var keyInfo = clause as KeyInfoNode;
+                    XmlElement keyInfoEl = keyInfo.GetXml();
+                    Assert.Equal("RetrievalElement", keyInfoEl.LocalName);
+                    Assert.Equal("http://www.w3.org/2000/09/xmldsig#", keyInfoEl.NamespaceURI);
+                    Assert.Equal(1, keyInfoEl.Attributes.Count);
+                    Assert.Equal("URI", keyInfoEl.Attributes[0].Name);
+                    Assert.Equal(retrievalElementUri, keyInfoEl.GetAttribute("URI"));
+                }
+                else if (clause is KeyInfoX509Data)
+                {
+                    pathsCovered |= 1 << 4;
+
+                    var x509data = clause as KeyInfoX509Data;
+                    Assert.Equal(1, x509data.Certificates.Count);
+                    X509Certificate cert = x509data.Certificates[0] as X509Certificate;
+                    Assert.NotNull(cert);
+                    Assert.Equal(Convert.FromBase64String(x509cert), cert.GetRawCertData());
+                }
+                else
+                {
+                    Assert.True(false, $"Unexpected clause type: {clause.GetType().FullName}");
+                }
+            }
+
+            // 0x1f = b11111, number of ones = 5
+            Assert.Equal(pathsCovered, 0x1f);
```

Comment:
Does this really need to be one mondo test, or can/should it be broken up into individual tests?